core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -659,13 +682,20 @@ class GroupMetadataManager(val brokerId: Int,
       config.offsetsTopicNumPartitions
   }
 
-  private def getMessageFormatVersionAndTimestamp(partition: Int): (Byte, Long) = {
+  /**
+   * Check if the replica is local and return the message format version and timestamp
+   *
+   * @param partition   Partition of GroupMetadataTopic
+   *
+   * @return            Option[(MessageFormatVersion, TimeStamp)] if replica is local, None otherwise
+   */
+  private def getMessageFormatVersionAndTimestamp(partition: Int): Option[(Byte, Long)] = {
     val groupMetadataTopicAndPartition = new TopicAndPartition(Topic.GroupMetadataTopicName, partition)
-    val messageFormatVersion = replicaManager.getMessageFormatVersion(groupMetadataTopicAndPartition).getOrElse {
-      throw new IllegalArgumentException(s"Message format version for partition $groupMetadataTopicPartitionCount not found")
+    val messageFormatVersionOpt = replicaManager.getMessageFormatVersion(groupMetadataTopicAndPartition)
```

Comment:
Not really. Just makes it explicit but I agree it can be made less verbose as you have said. :)